api/src/main/java/io/smallrye/dux/Dux.java

```diff
@@ -0,0 +1,104 @@
+        // mstodo test for multiple config providers!!
+        Optional<DuxConfigProvider> highestPrioConfigProvider = configs.stream()
+                .map(ServiceLoader.Provider::get)
+                .max(Comparator.comparingInt(DuxConfigProvider::priority));
+
+        DuxConfigProvider configProvider = highestPrioConfigProvider.orElseThrow(
+                () -> new IllegalStateException("No DuxConfigProvider found"));
+
+        for (ServiceConfig serviceConfig : configProvider.getDuxConfigs()) {
+            ServiceDiscoveryConfig serviceDiscoveryConfig = serviceConfig.serviceDiscovery();
+            if (serviceDiscoveryConfig == null) {
+                throw new IllegalArgumentException(
+                        "No service discovery not defined for service " + serviceConfig.serviceName());
+            }
+            String serviceDiscoveryType = serviceDiscoveryConfig.type();
+            if (serviceDiscoveryType == null) {
+                throw new IllegalArgumentException(
+                        "Service discovery type not defined for service " + serviceConfig.serviceName());
+            }
+            ServiceDiscoveryProvider serviceDiscoveryProvider = serviceDiscoveryProviders.get(serviceDiscoveryType);
+            if (serviceDiscoveryProvider == null) {
+                throw new IllegalArgumentException("ServiceDiscoveryProvider not found for type " + serviceDiscoveryType);
+            }
+
+            ServiceDiscovery serviceDiscovery = serviceDiscoveryProvider.createServiceDiscovery(serviceDiscoveryConfig);
+            serviceDiscoveries.put(serviceConfig.serviceName(), serviceDiscovery);
+
+            LoadBalancerConfig loadBalancerConfig = serviceConfig.loadBalancer();
+            if (loadBalancerConfig == null) {
+                // no load balancer, maybe someone intends to use service discovery only, ignoring
+                // TODO: log debug sth
+            } else {
+                String loadBalancerType = loadBalancerConfig.type();
+                LoadBalancerProvider loadBalancerProvider = loadBalancerProviders.get(loadBalancerType);
+                if (loadBalancerProvider == null) {
+                    throw new IllegalArgumentException("No LoadBalancerProvider for type " + loadBalancerType);
+                }
+
+                LoadBalancer loadBalancer = loadBalancerProvider.createLoadBalancer(loadBalancerConfig, serviceDiscovery);
+                loadBalancers.put(serviceConfig.serviceName(), loadBalancer);
+            }
+        }
+    }
+
+    private <T extends ElementWithType> Map<String, T> getAll(Class<T> providerClass) {
+        ServiceLoader<T> providers = ServiceLoader.load(providerClass);
+        return providers.stream().map(ServiceLoader.Provider::get)
+                .collect(Collectors.toMap(ElementWithType::type, Function.identity()));
+    }
+
+    private static final Dux dux = new Dux();
+
+    public static Dux getInstance() {
+        return dux;
+    }
+}
```

Comment:
I think we should add some unit tests in the api module, wdyt? May be in a differente PR?